### PR TITLE
add remove_redundant_generators for sspz

### LINKED ISSUE #3005: Simplifations of sspz
The example added [here](https://github.com/JuliaReach/LazySets.jl/pull/2990/files#diff-841de8744f3df2e459762fdec470f5b7268493c42b131e8a1ecc12a1884c59f7R57) highlights that we should have a simplification function such that:

- Zero columns in G can be eliminated.
- Zero columns in E can be associated with the center for the corresponding generator.
- Repeated columns in E can be grouped together by summing the corresponding generators.

(This simplification is exact btw)

cc @lucaferranti

### PR DESCRIPTION
**Pull request by lucaferranti**
fixes #3005

review: please check I didn't misunderstand / oversee something in the algorithm/tests :)

### CODE REVIEW

File: src/Sets/SimpleSparsePolynomialZonotope.jl [hunk 1]
```diff
@@ -258,3 +258,41 @@ function quadratic_map(Q::Vector{MT}, S::SimpleSparsePolynomialZonotope) where {
     end
     return SimpleSparsePolynomialZonotope(cnew, Gnew, Enew)
 end
+
+"""
+    remove_redundant_generators(S::SimpleSparsePolynomialZonotope)
+
+remove redundant generators from the representation of `S`. Let `G` the generator matrix and
```
Comment (schillic) [inline src/Sets/SimpleSparsePolynomialZonotope.jl:265]:
```suggestion
Remove redundant generators from the representation of `S`. Let `G` be the generator matrix and
```

**Review by schillic — APPROVED** [review]

**Comment by mforets** [comment]
The algorithm and tests lgtm! 👌

File: src/Sets/SimpleSparsePolynomialZonotope.jl [hunk 2]
```diff
@@ -258,3 +258,31 @@ function quadratic_map(Q::Vector{MT}, S::SimpleSparsePolynomialZonotope) where {
     end
     return SimpleSparsePolynomialZonotope(cnew, Gnew, Enew)
 end
+
```
[the remark this answers is not in the document]
Comment (lucaferranti) [reply src/Sets/SimpleSparsePolynomialZonotope.jl:261]: I added one in the latest commit, are you looking at an outdated version?

[the remark this answers is not in the document]
Comment (mforets) [reply src/Sets/SimpleSparsePolynomialZonotope.jl:261]: I see it now :+1:  yes maybe i was looking at an outdated version.

[the remark this answers is not in the document]
Comment (mforets) [reply src/Sets/SimpleSparsePolynomialZonotope.jl:261]: I made minor tweaks in the last commit.

**Review by mforets — APPROVED** [review]

**Comment by lucaferranti** [comment]
@mforets apparents the checks for the docs failed

**Comment by lucaferranti** [comment]
see https://github.com/JuliaReach/LazySets.jl/runs/7378684833?check_suite_focus=true

**Comment by mforets** [comment]
i guess it's because i merged before it had finished. we can check again in https://github.com/JuliaReach/LazySets.jl/pull/3010

**Pull request merged**